verilog/tools/ls/symbol-table-handler.cc

```diff
@@ -0,0 +1,177 @@
+// Copyright 2021 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "verilog/tools/ls/symbol-table-handler.h"
+
+#include <filesystem>
+
+#include "common/strings/line_column_map.h"
+
+namespace verilog {
+
+static const std::string fileschemeprefix = "file://";
+
+bool LSPUriToPath(absl::string_view uri, std::string *path) {
+  auto isfileuri = absl::StartsWith(uri, fileschemeprefix);
+  if (!isfileuri) {
+    return false;
+  }
+  if (!path) {
+    return false;
+  }
+  std::string res{uri.substr(fileschemeprefix.size())};
+  *path = res;
+  return true;
+}
+
+bool PathToLSPUri(absl::string_view path, std::string *uri) {
+  if (!uri) {
+    return false;
+  }
+  std::filesystem::path p = std::string(path);
+  *uri = fileschemeprefix + std::filesystem::absolute(p).string();
```

Comment:
For all string concatenation it is better to use `absl::StrCat()` (it works with string views and other random types, but also is usually faster)